frameworks/CSharp/appmpower/src/HttpApplication.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using appMpower.Kestrel;
+using PlatformBenchmarks;
+
+namespace appMpower
+{
+   public class HttpApplication : IHttpApplication<IFeatureCollection>
+   {
+      private readonly static AsciiString _plainText = "Hello, World!";
+
+      public IFeatureCollection CreateContext(IFeatureCollection featureCollection)
+      {
+         return featureCollection;
+      }
+
+      public async Task ProcessRequestAsync(IFeatureCollection featureCollection)
+      {
+         var request = featureCollection as IHttpRequestFeature;
+         var httpResponse = featureCollection as IHttpResponseFeature;
+         var httpResponseBody = featureCollection as IHttpResponseBodyFeature;
+
+         PathString pathString = request.Path;
+
+         if (pathString.HasValue)
+         {
+            int pathStringLength = pathString.Value.Length;
+            string pathStringStart = pathString.Value.Substring(1, 1);
+
+            if (pathStringLength == 10 && pathStringStart == "p")
+            {
+               PlainText.Render(httpResponse.Headers, httpResponseBody.Writer, _plainText);
+               return;
+            }
+            else if (pathStringLength == 5 && pathStringStart == "j")
+            {
+               Json.RenderOne(httpResponse.Headers, httpResponseBody.Writer, new JsonMessage { message = "Hello, World!" }, new JsonMessageSerializer());
```

Comment:
`new JsonMessageSerializer()` looks idempotent and everything is passed via parameter; so could hoist this class to a readonly static rather than allocating each time?

Reply:
Thanks for the review Ben. This recurring suggestion is a good one. As I am new on github since April and this is my first TechEmpower Benchmark pull request, can you just let me know how I best do the change ? Do I start locally in my appMpowerv1 branch, commit there, push to my forked repository and then do a new pull request to this repository ? Just to be sure: I assume your suggestion is like adding  
 
private readonly static WorldSerializer _worldSerializer = new WorldSerializer(); 

at the top of HttpApplication.cs and using that one ? 
Luc

Reply:
> Just to be sure: I assume your suggestion is like adding
> 
> private readonly static WorldSerializer _worldSerializer = new WorldSerializer();

Yep

> Do I start locally in my appMpowerv1 branch, commit there, push to my forked repository and then do a new pull request to this repository ?

If the PR is still open (e.g. this one) you just need to push to same fork and it will automatically update, you don't need to do another pull request.